src/context.ts

```diff
@@ -94,7 +95,13 @@ export class Context {
 
         return instances;
       })
-      .filter(instance => checks.every(check => check(instance.value)))
+      .filter(instance => {
+        if (options.includeOptionals === 'ignoreUndefined') {
+          return checks.every(check => check(instance.value)) && instance.value !== undefined;
+        } else {
+          return checks.every(check => check(instance.value));
+        }
+      })
```

Comment:
This piece of code should stay unedited and the update should be made upper in the various checks definition.